pages/favorites.jsx

```diff
@@ -0,0 +1,80 @@
+import useLocalStorageState from "use-local-storage-state";
+import ActivityCard from "../components/ActivityCard";
+import styled from "styled-components";
+import useSWR from "swr";
+import { useEffect, useState } from "react";
+
+export default function Favorites() {
+  const [favoriteIds, setFavorites] = useLocalStorageState("favorites", {
+    defaultValue: [],
+  });
+
+  const {
+    data: activities,
+    error,
+    isLoading,
+  } = useSWR(
+    favoriteIds.length > 0 ? ["/api/activities/favourites", favoriteIds] : null,
+    async ([url, ids]) => {
+      const response = await fetch(url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ ids }),
+      });
+      if (!response.ok) {
+        throw new Error("Failed to fetch favorite activities");
+      }
+      return response.json();
+    }
+  );
+
+  const handleFavoriteToggle = (activityId) => {
+    const newFavorites = favoriteIds.filter((id) => id !== activityId);
+    setFavorites(newFavorites);
+
+    activities.filter((activity) => activity._id !== activityId);
+  };
+
+  if (isLoading) {
+    return <Title>Loading...</Title>;
+  } else if (error) {
+    return <Title>Error</Title>;
+  }
```

Comment:
```suggestion
  if (error) {
    return <Title>Error</Title>;
  }
```

remove `if (isLoading)` as `isLoading` short-circuits your UI wiping out the rest of the component and making it appear like it's "refetching".